src/ser/mod.rs

```diff
@@ -1329,6 +1337,30 @@ impl<'a, W: fmt::Write> ser::SerializeStruct for Compound<'a, W> {
 
         if !self.ser.compact_structs() {
             self.ser.indent()?;
+
+            if let Some((ref config, _)) = self.ser.pretty {
+                let mut iter = self.ser.field_memory.iter();
+
+                let init = iter.next().and_then(|name| config.meta.fields().get(name));
+                let field = iter
```

Comment:
Would it be possible to not have this loop for every compound? What I'm imagining is a stack of borrows into the metadata instead of a stack of field names, so that when we serialise a field, we already know if it has metadata and don't have to search any further

Reply:
If we're talking about storing an `Option<&Meta>` in the Serializer instead of a stack of field names, it'd be self-referential, which the borrow checker will not particularly like.

Reply:
It would be an Vec<&Fields>, but your point stands … is there another way we could avoid the quadratic runtime here?

Reply:
We could avoid borrowing by mutating the PrettyConfig in place during serialization.

In serialize_field:
Before serialization:
- Remove the field we are entering from the config.
- Swap the config's current field with the removed field.
During serialization:
- The field is directly accessed, no further processing.
After serialization:
- Swap the config's current field with the original field.
- Insert the removed field back into the config.

This moves the head of the config so it always points directly at the field we're currently serializing. Performance-wise, it'd just be swapping of `Field` (<100 bytes).

Reply:
That sounds like a good plan to me